%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
\usepackage{graphicx}
\usepackage{setspace}
\usepackage[T1]{fontenc}
\usepackage{longtable}
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%\def\includegraphic{}
%\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Mimoza: Web-Based Semantic Zooming and Navigation in Metabolic Networks}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   corref={aff1},                       % id of corresponding address, if any
   %noteref={n1},                        % id's of article notes, if any
   email={anna.zhukova@inria.fr}   % email address
]{\inits{AZ}\fnm{Anna} \snm{Zhukova}}
\author[
   addressref={aff1},
   email={david.sherman@inria.fr}
]{\inits{DJS}\fnm{David J} \snm{Sherman}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Inria/Universit\'e Bordeaux/CNRS joint project-team MAGNOME}, % university, etc
  \street{351, cours de la Lib\'{e}ration},                     %
  \postcode{F-33405}                                % post or zip code
  \city{Talence},                              % city
  \cny{France}                                    % country
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract

\parttitle{Background}
The complexity of genome-scale metabolic models makes them quite difficult for human users to read, since they contain thousands of reactions that must be included for accurate computer simulation. % For example, the metabolic network of the bacterium \emph{E. coli} contains 2\,168 reactions, the \emph{yeast 7} metabolic network model of \emph{S.~cerevisiae} includes 2\,352 reactions, and \emph{recon 2}, a global human metabolism reconstruction, has 7\,440 reactions. 
Interestingly, hidden similarities between groups of reactions can be discovered, and generalized to reveal higher-level patterns.

\parttitle{Results}
The web-based navigation system Mimoza allows a human expert to explore metabolic network models in a semantically zoomable manner: The most general view represents the compartments of the model; the next view shows the generalized versions of reactions and metabolites in each compartment; and the most detailed view represents the initial network with the generalization-based layout (where similar metabolites and reactions are placed next to each other). It allows a human expert to grasp the general structure of the network and analyze it in a top-down manner.

\parttitle{Conclusions}
Mimoza can be installed standalone, or used on-line at http://mimoza.bordeaux.inria.fr/, or installed in a Galaxy server for use in workflows.
Mimoza views can be embedded in web pages, or downloaded as COMBINE archives.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{metabolic modeling}
\kwd{visualization}
\kwd{model generalization}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
%The Background section should be written in a way that is accessible to researchers without specialist knowledge in that area and must clearly state - and, if helpful, illustrate - the background to the research and its aims. It should clearly described the relevant context and the specific issue which the software described is intended to address.

Semantic generalization of metabolic network models~\cite{Zhukova2014} is a theoretical method designed to aid users in understanding complex models. Generalization identifies and groups similar metabolites and similar reactions in the network.
Applied to different models, it can bring them to the same level of abstraction so that they can be compared.
To explore the opportunities of the method we implemented it as a practical tool.

The \emph{zooming user interface} (ZUI)~\cite{Bederson1998} paradigm has proven to be a powerful tool for representation of data at different scales. It is being adopted for various domains of applications, including cartographic~\cite{Nivala2008}, exploratory data visualization~\cite{Roberts2005}, collaborative interfaces~\cite{Laufer2011} and biological data~\cite{Pook1998, Hu2007}. The challenge is how to use ZUI-based visualization for semantic generalization of metabolic networks.

\subsection*{Metabolic network reconstruction and infrastructure}
There is a conflict between the level of detail of metabolic models needed for computer simulation and the one that can be easily analyzed by a human curator: Genome-scale metabolic models include thousands of reactions that may participate in organism's metabolism (e.g.,  2\,251 reactions in the metabolic network of the bacterium \emph{E. coli}~\cite{Orth2011}, 2\,352 reactions in the \emph{yeast 7} metabolic network model of \emph{S.~cerevisiae}~\cite{Aung2013}, 7\,440 reactions in \emph{recon 2},  a global human metabolism reconstruction~\cite{Thiele2013}), while human experts understand best small-sized networks, containing up to hundreds of nodes~\cite{VonLandesberger2011,Herman2000}.

Metabolic network reconstruction can address various objectives. Examples include creation of a model for a new organism from its genomic data and a reference model for a similar organism; creation of a larger-scale model by combining several models of different aspects of organism's metabolism; improving an existing model by incorporating new data and new expertise. To accomplish these objectives the following tasks are used (see Figure~\ref{fig:workflow}). 

\paragraph{Inference}

%% Reconstruction
The metabolic network reconstruction process is becoming more advanced, and there now exist various tools for semi-automatic model inference, e.g. PathwayTools~\cite{Karp2002}, CoReCo~\cite{Pitkanen2014}, SuBliMinaL~\cite{Swainston2011} (see~\cite{Hamilton2014} for a review).

%% Storage
Starting from a model for a similar organism or a collection of pathways, and genomic data, they produce a draft model for the target organism. Existing metabolic models can be found in several resources, including Biomodels Database~\cite{Li10}, BIGG~\cite{Schellenberger2010}, JWS online~\cite{Snoep2003}. KEGG~\cite{Kanehisa12} and Reactome~\cite{Milacic2012, Croft2013} provide an extensive collection of pathways. 

%% Representation
Models are stored and shared using established formats, such as SBML~\cite{Hucka2003}, SBGN~\cite{LeNovere2009}, CellML~\cite{Lloyd2004}. A model represented in these formats can be further enriched with the knowledge from biological databases and ontologies, e.g. ChEBI~\cite{deMatos10}, Uniprot~\cite{TheUniProtConsortium2014}, by annotating elements of the models (such as metabolites, reactions) with appropriate identifiers. Further in this manuscript we will consider metabolic models in SBML format.

Although automatic model inference tools and genomic comparison methods are becoming steadily more sophisticated, they may still leave gaps in the model or add erroneous reactions. The intrinsic and extrinsic correctness of the model should be checked during the phases of analysis and curation.


\paragraph{Curation and analysis}
The inferred draft network needs to be refined during several iterations of analysis, curation and improvement~\cite{Thiele2010,Swainston2011}. The goal of the \emph{model analysis} is to verify that the model does not contain inner contradictions and errors, e.g., that the network is connected; the transport reactions between compartments are well defined; the reactions are chemically balanced, etc. Various model analysis tools, e.g, FASTGAPFILL \cite{Thiele2014} for gap filling, CellNetAnalyser~\cite{Klamt07} for for finding dead ends and blocked reactions, SuBliMinaL Toolbox~\cite{Swainston2011} for reaction balancing, can facilitate model analysis; but human expert's knowledge on organism's metabolism still plays an important role.

The \emph{curation} is needed to %check the extrinsic correctness of the model, \ie to 
ensure that the experts' knowledge is stored in the model in an appropriate way, such as that the metabolites and reactions are annotated with the identifiers from the corresponding knowledge bases and additional important information, making their names unambiguous for the audience apart from the model creators, as well as for the machines; that all the known processes that need to be part of the model are present and correct, etc.

The processes of curation and analysis can be repeated several times, iteratively improving the draft model, till the satisfiable level of quality is achieved.

The curation by a human expert requires a means of splitting genome-scale models into smaller units that can be checked and analyzed independently. At a higher level, appropriate levels of abstraction need to be found to allow experts to compare whole genome networks. Good model visualization tools are also required.

\paragraph{Simulation}
The improved model, created during the iterations of curation and analysis, %can be used in several ways (see processes with the light red background in Figure~\ref{fig:workflow}). It 
can be used for computer simulation to obtain numerical results. We do not exploit simulation in this manuscript.

\paragraph{Exploration}
The model can also be used for knowledge-oriented exploration to obtain new knowledge about the processes happening in the organisms' metabolism, and the relationships between them, e.g., the ``redundancy'' of the model: discovery of similar reactions, and alternative pathways. 
Means of splitting genome-scale models into smaller units, appropriate levels of abstraction and good model visualization tools are as important for model exploration task as they are for curation.

\paragraph{Comparison and combination}
Model comparison and combination is another important task. Possible scenarios include comparison to a different model of the same organism, with potential merging into a new, more complete, model; comparison of a model of a healthy organism to the one of a metabolism suffering from a disease to discover disease-specific metabolic adaptations.
A genome-scale model can be created by combining several smaller models, describing different metabolic processes in a species~\cite{Schulz2006}, where model comparison is needed to detect overlaps. Such a model can be used as a draft model, and will need to undergo the analysis and curation phase.
Finally, a group of models for related species can be compared and combined to produce a concise representation of their common metabolism, to study the common properties of a group, as well as the organism-specific adaptations.
 
There exist various software facilitating model merging, e.g.,  semanticSBML~\cite{Krause2010}, OREMPdb~\cite{Umeton2012}, PathCase-SB Model Composition Tool~\cite{Coskun2013}, but all of them require human expert's intervention in cases when the models to be merged are incompatible or contradict to each other, as well as for better discovery of common parts. Thereby, after the creation, the combined model becomes a draft and should in its turn undergo the analysis and curation cycle.

By combining these modeling tasks into workflows, as in Figure~\ref{fig:workflow}, one can accomplish the modeling objectives listed above.

At least three of the aforementioned tasks (curation, exploration, comparison) require the intervention of a human expert, and thus require methods of dealing with the complexity of the models, e.g., by splitting them into smaller modules, by defining different levels of abstraction, and by visualization.


\subsection*{Existing visualization approaches}
% Desktop
There exist various modeling tools for metabolic networks that also support visualization. Desktop tools include CellDesigner~\cite{Funahashi2008}, VANTED~\cite{Rohn2012}, and Cytoscape~\cite{Smoot2011}. They produce reasonably good visualizations of small networks (up to hundreds of reactions), but become cluttered at the genome-scale level, making the visualization unreadable. For comparison, the winner of the best SBGN map competition~\cite{SBGN}, the ER Stress response~\cite{Groenendyk2010} map,  was created manually in CellDesigner.

% Web-based
Web-based tools allowing for metabolic network visualization are also emerging.  JWS online~\cite{Snoep2003}, for example, provides a mechanism for network visualization using a force-directed layout algorithm~\cite{Fruchterman1991, Tamassia:2007:HGD:1202383}. It also encounters the aforementioned issues and thus is not capable of providing a readable representation for large networks.  

MetDraw~\cite{Jensen2014} is an online tool for genome-scale metabolic model visualization, that makes use of decomposition of the model into compartments and pathways (if the pathway information is present in the model as a \emph{subsystem} annotation of reactions) and duplication of minor metabolites. Metabolite duplication reduces clutter, but the huge number of reactions in the compartments of some models and missing \emph{subsystem} annotations, makes the visualization consume too much space and do not allow a user to grasp the essential structure of the network.

% ZUI
Due to the huge numbers of reactions and of metabolites participating in multiple reactions, we have an uncomfortable choice between either many edge crossings in an automatic visualization of a genome-scale network, or over-duplication of various metabolites making the essential parts of the network disconnected and the visualization too large to grasp. Therefore an approach different to a simple graph layout algorithm is necessary. ZUIs, which can change the size and nature of the content displayed at different zoom levels, provide a pertinent alternative. Two main types of magnification can be considered: \emph{geometric zooming}, in which a region of the network is enlarged; and \emph{semantic zooming}, in which additional properties are introduced with enlargement~\cite{Hu2007}.

Semantic zooming was first introduced for biological data visualization in 1988 with Zomit~\cite{Pook1998}, a generic application programming interface for developing  servers for zoomable navigation and visualization, and illustrated with an example of ZoomMap, a prototype browser for HuGeMap human genome database~\cite{Barillot1998}. The work by Jianlu and Laidlaw~\cite{Jianu2013} evaluates geometric zooming with the Google Maps interface on five examples (a gene co-regulation visualization, a gene expression heatmap viewer, a genome browser, a protein interaction network, and  neural projections), and describes a positive feedback provided by both domain experts and less experienced users. Another example of a Google Maps-based ZUI is X:map~\cite{Yates2008}, a genome annotation database that supports zoomable data browsing. It does not use semantic zooming, but allows for showing/hiding layers with additional information (EST and GenScan predictions).

There exist several web-based tools that include a zoomable representation of metabolic networks. Genome Projector~\cite{Arakawa2009} is a zoomable genome map with multiple views, including a pathway map. The pathway map is based on the Roche Biochemical Pathway wall chart available from the ExPASy proteomics server~\cite{Gasteiger2003}. The Roche Biochemical Pathway wall chart has a large size and shows the collection of biochemically known molecules, enzymes and reactions. Genome Projector provides a geometric zooming on the map and overlay layers to highlight reactions present in the organism of interest. The list of organisms is fixed to 320 bacterial genomes. The full Roche Biochemical Pathway map with the fixed layout is always shown, but only the reactions of interest (corresponding to the chosen organism) are highlighted.

NaviCell~\cite{Kuperstein2013} is a web environment that permits exploiting large maps of molecular interactions, including metabolic maps. It allows users to create their own maps, but does not provide a solution to the problem of huge network layout. The map creation is not fully automatic: The user must create a map in CellDesigner, export it as an image and partly manually edit it in a graphical designer to produce intermediate views (possibly with different level of details for semantic zooming). In addition, NaviCell permits a user to split the map into submaps called modules.

Another web-based tool, the Cellular Overview~\cite{Latendresse2011} creates interactive diagrams for metabolic maps of organisms in the BioCyc database~\cite{Caspi2012}. It is pathway-oriented, and supports only geometric zooming. Another drawback is that it does not show the compartmentalization.

The Reactome pathway database~\cite{Milacic2012, Croft2013} browser provides a zommable visualization of manually curated pathways for 19 organisms. It has two semantic zoom levels: a general representation of organism's pathways (nodes represent pathways, the edges connect the related ones); and submaps showing the details of each of the pathways, including compartmentalization. Several levels of geometric zoom are available on both semantic zoom levels. Reactome is pathway-oriented. Inside each pathway the layout is fixed: reactions, metabolites, and compartments common to two organisms have the same layout in corresponding representations. On the other hand, the positions and sizes of compartments might differ between pathways of the same organism.

None of the ZUI tools for metabolic map representation described above, except for NaviCell, allow users to input their own models. Moreover, as these examples show, not only geometric zoom but also model decomposition and semantic zoom are important for multi-level visualization of huge models. At the general level, the network needs to be decomposed into several meaningful modules (such as compartments, pathways). If after such a decomposition the model remains complicated (e.g. the mitochondrial compartment of the yeast consensus model~\cite{Herrgard2008} containing 230 reactions), a further decomposition is required. We address these issues below by combining model generalization with a ZUI.


\section*{Implementation}
% This should include a description of the overall architecture of the software implementation, along with details of any critical issues and how they were addressed.

\subsection*{Choosing zoom levels}
We address the problem of large-scale metabolic model visualization by combining meaningful decomposition into modules with automatic multi-level abstraction. Decomposition is performed in the following way: The network is first split into compartments; then the model generalization method is applied to each compartment to detect the generalized modules. Thereby, the most appropriate is to adopt 3 levels of semantic zooming:

\begin{enumerate}
\item The most abstract level represents compartmentalization of the network, and focuses on such questions as: Are all the compartments present? Are they well connected by transport reactions?

This level shows the compartments of the model, the transport reactions between them, and other reactions happening inside the cytoplasm.
If the model does not describe compartments, this level will be missing.

\item The second level shows the modules inside each of the compartments. The questions to be addressed at this level include: Are all the essential processes present? Is the structure of each process correct? Are there organism-specific adaptations of the structure?

We use our knowledge-based generalization method to identify the modules inside the compartments. It detects similar metabolites and reactions and clusters them together to represent them as generalized metabolites and reactions with the same structure (numbers of consumed and produced metabolites). The generalized representation keeps the essential structure of the network while hiding the details.

If no similar metabolites/reactions can be detected by the generalization method (due to the model structure or to missing ChEBI metabolite annotations), this level will be missing.

\item The most detailed level is intended for computer simulation and represents the inner structure of each of the modules with all the metabolites, reactions and their kinetics, stoichiometries and constraints.

Our method places similar metabolites and reactions (detected at level 2) next to each other, thus simplifying the analysis of their presence.

\end{enumerate}

\subsection*{Model generalization}
The metabolic model generalization method~\cite{Zhukova2014}, which we recall here, groups similar metabolites and reactions in the network based on its structure and the knowledge extracted from metabolite ontologies. 

To make metabolite grouping semantically meaningful, an ontology describing hierarchical relationships between biochemical entities is used. Each metabolite can be generalized up to one of its ancestors in the ontology. We use the \textit{ChEBI} ontology, as it is the de facto standard for metabolite annotation in metabolic networks. If a ChEBI annotation for a metabolite is not present in the model, the method attempts to automatically deduce it by comparing metabolite's name to ChEBI terms' names and synonyms. 

Reactions that share the same generalized reactants and the same generalized products, are considered equivalent and are factored together into a generalized reaction. 

The appropriate level of abstraction for metabolites and reactions is defined by the network itself as the most general one that satisfies two restrictions: 
\begin{enumerate}
 \item \emph{Stoichiometry preserving restriction}: metabolites that participate in the same reaction cannot be grouped together;
 \item \emph{Metabolite diversity restriction}: metabolites that do not participate in any pair of similar reactions are not grouped together (as there is no evidence of their similarity in the network).
\end{enumerate}

Overall, the generalization method is composed of three modules: 
\begin{enumerate}
\item \emph{Aggressive reaction grouping} based on the most general metabolite grouping (defined by ChEBI), in order to generate reaction grouping candidates;
\item \emph{Ungrouping of some metabolites and reactions} to correct for violation of the stoichiometry preserving restriction;
\item \emph{Ungrouping of some metabolites} (while keeping the reaction grouping intact) to correct for violation of the metabolite diversity restriction.
\end{enumerate} 

For instance, \textit{(S)-3-hydroxydecanoyl-CoA}, \textit{(S)-3-hydroxylauroyl-CoA} and \textit{(S)-3-hydroxytetradecanoyl-CoA} have a common ancestor \textit{hydroxy fatty acyl-CoA} in \textit{ChEBI}. They can be grouped and generalized into \textit{hydroxy fatty acyl-CoA}, if in the network there is no reaction whose stoichiometry would be changed by such a generalization (stoichiometry preserving restriction), and exist similar reactions that consume or produce them (metabolite diversity restriction).

The method is available as a python library~\cite{Metamogen} that operates on models in \textit{SBML}~\cite{Hucka08} format. It takes an SBML file of level~2 or~3 (any version) and produces an SBML level~3 version~1 file with groups extension~\cite{Hucka2012} that contains the initial model plus groups for all non-trivial similar metabolite and reaction sets (see Figure~\ref{fig:groups}).

The compression that can be achieved with the model generalization method depends on the model structure and on how well the model is annotated with the ChEBI ontology (as the metabolites lacking ChEBI annotations are not generalized). Table~\ref{tbl:gen} shows the results of the application of the model generalization method to 269 metabolic models from Path2Model project~\cite{Buchel2013}. All those models are genome-scale, the average number of reactions per model is 2\,879. The average compression ratio ($\frac{number\,of\,reactions\,in\,the\,initial\,model}{number\,of\,reactions\,in\,the\,generalized\,model}$) is 1.14.

% Ubiquitous metabolites are duplicated (to improve readability)
%We do not generalize ubiquitous (frequently occurring) metabolites, e.g. \textit{oxygen}, \textit{hydrogen}, \textit{water}, \textit{ATP}. Grouping metabolites increases the number of reactions they participate in, while these are already shared by many reactions and networks to such an extent that during visualization these metabolites are usually duplicated~\cite{Rohn2012} to improve readability.

\subsection*{Layers Layout}
To visualize a metabolic network we first represent it as a bipartite graph~\cite{Diestel2012} with two disjoint sets of nodes (metabolites and reactions), and edges that connect the reactions to their substrate and product metabolites. To achieve such a representation, we implemented a converter from SBML to TLP format, that is used by the Tulip graph visualization tool~\cite{Auber04}. TLP format stores nodes and edges of the graph, and associates each node and edge to a list of named attributes: standard ones, such as shape, size, color; and user-defined ones, such as, in our case, element type (compartment, reaction or metabolite), ChEBI identifier, group number, gene association, etc. The SBML-to-TLP converter is implemented in python, using libSBML library~\cite{Bornstein2008}, and is available as a part of Mimoza software. 

While layout of large graphs is widely studied~\cite{Unwin2006}, the correspondence between the layouts of different semantic zoom levels remains a hard task. To compute the layout for different semantic zoom levels we combine two different approaches.

\subsubsection*{Generalized model layout}
% General idea
In order to lay out the sub-networks corresponding to each of the compartments after the generalization, we use a combination of standard layout algorithms provided by Tulip. We divide the compartment graph into connected components (subgraphs in which any two nodes are connected to each other by undirected paths, and which are not connected to any additional nodes in the supergraph), using a method provides by Tulip. We then apply an appropriate layout algorithm on each of them. The results are combined together using the \emph{Connected Component Packing} algorithm (provided by Tulip), which places the components close to each other while removing the overlaps between them.

% Laying out a connected component
Depending on the nature of the connected component subgraph, we choose one of the following layout algorithms, provided by Tulip:
\begin{itemize}
\item \emph{Hierarchical Layout} for the components that contain no cycles (\emph{Sugiyama (OGDF)}~\cite{Sugiyama1981} algorithm, that has the complexity of $O(|V||E|)$ in time and of $O(|V| + |E|)$ in space);
\item \emph{Circular Layout} for the components with less than 100 nodes and less than 3 cycles (\emph{Circular (OGDF)}~\cite{Tamassia:2007:HGD:1202383}, with $O(|E|^2)$ time and space complexity);
\item \emph{Force-Directed Layout} for all the other components (\emph{$FM^3$ (OGDF)}~\cite{Hachul2005}, that has the asymptotic worst-case running time of $O(|V|log|V|+|E|)$ with linear memory requirements).
\end{itemize}

% Ubiquitous metabolites hadling
To avoid clutter we duplicate all the \emph{ubiquitous} metabolites (\textit{oxygen}, \textit{hydrogen}, \textit{water}, \textit{ATP}, etc.) before applying the layout algorithms, so that there is a copy of an ubiquitous metabolite for each reaction in which it is used. We then extract a subgraph, containing all but the ubiquitous metabolites, apply the combined layout on it, and then place the ubiquitous metabolites next to the reactions in which they participate. 

\subsubsection*{Generalization-based full model layout}
The layout for the full model is based on the corresponding generalized model's layout. To allow zooming into the generalized model, we keep the same coordinates as in the generalized model for the ubiquitous metabolites and the ungeneralized metabolites and reactions, and place similar metabolites or reactions next to each other inside the space used by the corresponding generalized metabolites or reactions in the generalized model. 

An edge in the generalized view might expand into several edges in the full-model view, for example, if it is a generalized edge connecting a generalized metabolite to a generalized reaction. The positions of the edges after such an expansion might slightly differ from the corresponding generalized one.

\subsubsection*{Node colors}

A different color is assigned to each generalized metabolite/reaction; and is propagated to the corresponding metabolites/reactions of the full model. Ubiquitous metabolites are colored grey. Mimoza's interface includes a checkbox that permits to hide/show ubiquitous metabolites .

\subsubsection*{Node sizes}

The size of the nodes depends on their nature: ubiquitous metabolites are smaller than the non-ubiquitous ones; a radius of a generalized metabolite/reaction is calculated as a sum of radiuses of the elements that it groups; compartment sizes are defined by the layouts of the elements inside them, so that the compartments are represented as minimal rectangles containing all the corresponding elements. All non-ubiquitous specific (i.e., not generalized) metabolites are of the same size; as well as all specific reactions.

\subsubsection*{Relative positions of compartments}
Metabolic models may include several compartments, nested into each other. For example, the \emph{peroxisome} compartment is surrounded by its \emph{membrane}, and contained in \emph{cytoplasm}; the \emph{cytoplasm} is part of the \emph{cell}, which is surrounded by the \emph{cell envelope}.

SBML allows to represent relative positions of the compartments in the model with an optional \emph{outside} tag. However, it is not available in all SBML levels, nor is widely used.

To be able to visualize the compartments correctly even for the SBML models lacking this information, we infer their relative positions from the Gene Ontology (GO)~\cite{Ashburner2000}. We associate each compartment with a term from the \emph{cellular component} branch of GO by using annotations in the model if they are present, or matching the compartments' names otherwise. We then use the \emph{part\_of} and \emph{is\_a} relationships between the terms in GO to infer relative compartment positions. If no term for a compartment could be found, it is placed on the outer-most level. 

\subsubsection*{SBML layout}
To store the calculated layout of the model elements we use the layout extension~\cite{Gauges2013} of SBML. It allows to store the coordinates and sizes of the metabolites, reactions and compartments in the model. The TLP-to-SBML layout converter is implemented in python and is available as a part of Mimoza software.
If the SBML model submitted by the user contains the layout information, our software uses it for nodes' positions. Therefore, it is possible to visualize a model with Mimoza, download the resulting SBML with layout annotations, edit it manually or with another software and then revisualize the updated version with Mimoza.
 
\subsection*{ZUI}

The zoomable interactive representation is achieved using Leaflet~\cite{Agafonkin}, a JavaScript library for interactive maps. 

We export elements of the network graph (compartments, metabolites and reactions) as map features in GeoJSON format~\cite{Butler} in order to store their coordinates and metadata (e.g. ChEBI annotations for metabolites). Figure~\ref{fig:geojson} shows an example of a reaction represented in GeoJSON format. The TLP-to-GeoJSON converter is implemented in python and is available as a part of Mimoza software. 

The GeoJSON objects are then added as layers to the map and rendered by Leaflet into clickable elements at corresponding zoom levels. We follow SBGN Process Description language convention~\cite{LeNovere2009} to choose the glyphs for model elements' representation: Metabolites are drawn as circles linked by edges to the reactions where they participate; reactions are represented as squares; compartments are drawn as rectangles.  When a user clicks on a map element a pop-up appears (see Figure~\ref{fig:pop_up}) showing its name, identifier and additional information, e.g. gene associations and formulas for reactions. 
 

\subsection*{Embedding}
After the visualization with Mimoza is done, we provide a link for embedding the view in another web page.

\subsection*{Download and distribution}
   
One can use Mimoza in three different ways:
\begin{enumerate}
\item As a standalone application.
All Mimoza code is open-source and can be downloaded from the project web page~\cite{Zhukovaa} and installed on a local server.
\item On the Mimoza web server.
Mimoza web server~\cite{Zhukovaa} lets one test visualization for smaller SBML models, with the possibility to download the result as a COMBINE archive~\cite{Bergmann2014}, including the SBML file with groups (to store the metabolite and reaction groupings) and layout (to store the element coordinates) extensions, GeoJSON files with the coordinates of model elements, and the HTML, CSS and JavaScript files that are needed to view the visualization in a browser.
\item As a Galaxy~\cite{Blankenberg2010} project tool, so that generation of Mimoza views can be included in a Galaxy workflow.
The Galaxy wrapper for Mimoza is available for download from the project web page.
\end{enumerate}

 
\subsection*{Pipeline}
The overall Mimoza pipeline contains 5 steps:
\begin{enumerate}
\item The user submits a model in SBML format (level~2 or~3, any version) via a web form.
\item If the model does not yet contain groups, it is generalized using the model generalization method, and the resulting SBML file (level~3 version~1 with groups extension) is made available to the user.
\item The SBML file with groups of similar metabolites and reactions is converted into a Tulip graph: metabolite nodes are connected by edges to the nodes of the reactions in which they participate. The generalized metabolites and reactions form quotient nodes. The Tulip graph is split into sub-graphs corresponding to different compartments, and layout algorithms are applied to them.
\item The compartment sub-graphs are exported in GeoJSON format and rendered by the Leaflet library into an interactive map that is represented to the user.
\item The result can either be browsed on the Mimoza web page directly, or downloaded as a COMBINE archive and embedded into a different website.
\end{enumerate}


\section*{Results and Discussion}
%The Results and Discussion may be combined into a single section or presented separately. They may also be broken into subsections with short, informative headings. In any case what should be described is the functionality of the software together with data on how its performance and functionality compare with and improve on functionally similar existing software. There should then be a discussion of the intended use of the software, and the benefits that are envisioned together, if possible, with an outline for the planned future development of new features.
To illustrate the use of Mimoza and compare it with other available ZUI tools, we visualized the yeast consensus genome-scale metabolic network model~\cite{Herrgard2008}. The result can be found at \url{http://mimoza.bordeaux.inria.fr/yeast4/comp.html?id=C_8}. 
Three semantic zoom levels (compartment view, the generalized peroxisome compartment and the detailed view) corresponding to the part of the model representing $\beta$-oxidation of fatty acids~\cite{Metzler01} are shown in Figure~\ref{fig:zoom_levels}. The first level (bottom) shows the peroxisome compartment, and the reactions happening inside the cytoplasm; the second level (middle) shows the generalized structure of the peroxisome, the main processes happening in it; the most detailed level (top) represents the complete model, placing semantically similar metabolites and reactions next to each other.

We visualized the same model using MetDraw with no manual adjustments. The resulting SVG file\footnote{MetDraw -- http://www.metdraw.com/metdraw/bc7df60221ba314c383b1bf6e7dad4c3056f92bb} has only one zoom level with lots of clutter, that does not allow one to see the structure of the network.

Cellular Overview does not allow one to visualize a model provided by a user, but has a map of metabolism of \emph{Saccharomyces cerevisiae}.\footnote{Cellular Overview -- http://biocyc.org/overviewsWeb/celOv.shtml} It has a clear non-overlapping representation of various pathways present in the model, but does not show the compartmentalization. It is not automatic and is pathway-oriented, thus is not suitable for models having no pathway metadata. The zoom-in shows additional labels but all the metabolites and reactions are present at all the levels, making the elements at the most general level very small and hard to analyze.

Mimoza especially targets draft models during curation, allowing one to visualize them fully automatically and help to analyze them in a top-down manner, starting from the general structure and going down to the details. The generalized level differentiates it from other tools, since it shows both the overall network structure and fine-grain visualization in the most detailed level, automatically placing semantically similar metabolites next to each other. Mimoza does not depend on pathway information, automatically infers the relative compartment placement (e.g. places organelles inside the cytoplasm) and exploits a model in SBML format with ChEBI annotations for metabolites (if no annotations are present, it tries to infer them automatically based on metabolites' names).

Using generalization to compare two metabolic networks makes most sense if they have equivalent generalized nodes that can be placed in corresponding positions in the two layouts. Mimoza currently handles this correspondence between zoom levels of the same network, but does not guarantee such correspondence when two networks are laid out independently.
To meet this challenge, three strategies can be explored.
The first is to use constrained layout~\cite{Karl-FriedrichBohringer}, to impose the positions of key features in one network on the corresponding features of the second network.
The second is also to use constrained layout, with a catalog of standard positions for common motifs in generalized maps; for example, always lay out the generalized $\beta$-oxidation of fatty acids as a 4-step cycle, with standard positions for the generalized metabolites common for all the networks that incorporate $\beta$-oxidation.
The third strategy, which we are in the process of testing, is to learn a common layout by generalizing the union of the two networks. The idea is to combine the reactions into one set, run the generalization procedure on the union to fix the positions of the common features, then to build each of the layouts using only its own set of nodes. Each network layout only contains its own nodes, but the common nodes of the two networks will be in common positions.

% We also plan to make the tool more interactive giving a user an opportunity to change layout and annotations, run FBA, etc. 

Finally, the API of the Leaflet framework used for the interactive navigation can be used to integrate the maps with other web-based tools, such as annotation editors or simulation software.

% Allowing to combine several models as an input, and producing a common generalisation is another interesting direction.

\section*{Conclusions}
% %This should state clearly the main conclusions of the article and give a clear explanation of the importance and relevance of the software.
We have implemented Mimoza .... As a web-server it requires just a browser with enabled javascript. The library itself uses libSBML, leaflet, Tulip, etc. One needs to installed them as described in web page.

% Mimoza is an amazing tool that everyone was waiting for and can finally use and become happy!

\section*{Availability and requirements}
\textbf{Project name:} Mimoza\\
\textbf{Project home page:} http://mimoza.bordeaux.inria.fr\\
\textbf{Operating system(s):} Platform independent\\
\textbf{Programming language:} Python, JavaScript\\
\textbf{Other requirements:} JavaScript should be enabled in the web browser\\
\textbf{License:} CeCILL (GPL compatible)\\
\textbf{Any restrictions to use by non-academics:} no restrictions

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
The authors declare that they have no competing interests.

\section*{Author's contributions}
AZ and DJS conceived the study, AZ wrote the software, AZ and DJS wrote the manuscript.

\section*{Acknowledgements}
We would like to thank Dr. Antoine Lambert for helpful discussions about using the Tulip software as a library.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file
\bibliography{db}      % Bibliography file (usually '*.bib' )

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}


\begin{figure}[h!]
\centering
\includegraphics[scale=0.6]{workflow.png}
\caption{\csentence{Metabolic modeling workflow.} 
\label{fig:workflow}
The processes highlighted in yellow represent the \emph{model creation cycle}: The draft model is created by model inference tools based on models for similar organism, pathway and reaction information extracted from model repositories and pathway and reaction databases; it is then iteratively improved during the process of curation and analysis. The resulting model can in its turn be added to model repositories.
The processes highlighted in red show \emph{model usages}: simulation and knowledge-oriented exploration.
The processes highlighted in green describe \emph{comparison and combination of several models}. As the model creation cycle, they also include the curation and analysis stage.
The processes represented with the red arrows require human expert's intervention, and thus need good visualization tools, ways of splitting models into modules and different levels of abstraction.}
\end{figure}

 
      \begin{figure}[h!]
\centering
\includegraphics[scale=0.4]{groups.png}
\caption{\csentence{Representation of a generalized model in SBML level~3 version~1 format with groups extension.}
\label{fig:groups}
The output SBML file contains the initial model (including the lists of metabolites (called \emph{species} in SBML), reactions, etc.) plus the \emph{listOfGroups} section that represents non-trivial quotient metabolite and reaction sets. In the figure, a group representing a quotient metabolite set of \emph{hydroxy fatty acyl-CoAs} is shown; it includes \emph{(S)-3-hydroxydecanoyl-CoA} (s\_0045),  \emph{(S)-3-hydroxylauroyl-CoA} (s\_0051), etc. Each of those metabolites was previously declared in the \emph{listOfSpecies} section.}
\end{figure}

\begin{figure}[h!]
\includegraphics[scale=0.5]{figure2.png}
  \caption{\csentence{GeoJSON representation of a reaction}
  \label{fig:geojson}
      An SBML reaction is stored as a GeoJSON Point feature, with its layout coordinates encoded in the geometry section. The identifiers, labels and annotations, as well as the information on the reactant and product metabolites are stored as properties. The ``type'' property value specifies that this GeoJSON feature is a reaction.}
      \end{figure}
      
\begin{figure}[h!]
\includegraphics[scale=0.54]{figure3.png}
  \caption{\csentence{A reaction pop-up}
  \label{fig:pop_up}
      (right part) An example of a pop-up that opens when a user clicks on a reaction: It contains the information on the reaction name, identifier, reactant and product metabolites and their stoichiometries, as well as gene associations. (left part) Gene names are hyperlinks redirecting to the NCBI Gene database~\cite{NCBI}.}
      \end{figure}
      
  \begin{figure}[h!]
 \includegraphics[scale=0.41]{figure_main.png}
  \caption{\csentence{Three zoom levels}
  \label{fig:zoom_levels}  
      The most general zoom level (bottom) shows the peroxisome and a generalized transport reaction. The intermediate zoom (middle) shows the generalized processes inside the peroxisome compartment. The most detailed view (top) reveals the metabolites and reactions of the initial model.}
      \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}

\begin{center}
\begin{table}[h!]
\caption{Comparison of ZUIs for metabolic models.}\label{tbl:zui}
\begin{tabular}{c|c|c|c|c|c}
        \hline
         & & Semantic & User's & Automatic &  \\ 
        Tool name  & Fixed layout & zoom  & model & layout & Modules \\
        \hline
        Genome Projector & yes & no & no & - & no\\
        NaviCell & no & if created & yes & no & yes\\
        & & by user & & & \\
        Cellular Overview & yes & no & no & - & no\\
        Reactome & yes (same pathw. diff. org.) /& yes & no & - & yes\\
         & no (diff. pathw. same organism)& & &  & \\
        Mimoza & no & yes & yes & yes & yes \\
        \hline
\end{tabular}
\end{table}
\end{center}


\begin{center}
\begin{longtable}{c|c|c|c}
\caption{Performance of the model generalization method.}\label{tbl:gen}\\
        \hline
         & Number of & Number of  & Compression\\ 
        Model  & of reactions & of reactions  & rate\\ 
        & (initial model) & (generalized model)  & \\ \hline
\endfirsthead

\multicolumn{4}{c}%
{{\bfseries \tablename\ \thetable{} -- continued from previous page}} \\
\hline
         & Number of & Number of  & Compression\\ 
        Model  & of reactions & of reactions  & ratio\\ 
        & (initial model) & (generalized model)  & \\ \hline
\endhead

\hline \multicolumn{4}{r}{{Continued on next page}} \\ 
\endfoot

\hline \hline
\endlastfoot
BMID000000140205 & 4010 & 3469 & 1.16 \\
BMID000000140206 & 3631 & 3180 & 1.14 \\
BMID000000140207 & 889 & 801 & 1.11 \\
BMID000000140208 & 2366 & 1989 & 1.19 \\
BMID000000140209 & 4088 & 3576 & 1.14 \\
BMID000000140210 & 4597 & 3985 & 1.15 \\
BMID000000140211 & 2868 & 2538 & 1.13 \\
BMID000000140212 & 3887 & 3367 & 1.15 \\
BMID000000140213 & 3824 & 3329 & 1.15 \\
BMID000000140214 & 2678 & 2447 & 1.09 \\
BMID000000140215 & 1280 & 1147 & 1.12 \\
BMID000000140216 & 3319 & 2924 & 1.14 \\
BMID000000140217 & 3651 & 3204 & 1.14 \\
BMID000000140218 & 2209 & 1890 & 1.17 \\
BMID000000140219 & 2768 & 2279 & 1.21 \\
BMID000000140220 & 4162 & 3665 & 1.14 \\
BMID000000140221 & 2027 & 1921 & 1.06 \\
BMID000000140222 & 2550 & 2155 & 1.18 \\
BMID000000140223 & 2970 & 2632 & 1.13 \\
BMID000000140224 & 1723 & 1525 & 1.13 \\
BMID000000140225 & 2271 & 1944 & 1.17 \\
BMID000000140226 & 3108 & 2794 & 1.11 \\
BMID000000140227 & 4635 & 3955 & 1.17 \\
BMID000000140228 & 1582 & 1404 & 1.13 \\
BMID000000140229 & 3023 & 2670 & 1.13 \\
BMID000000140230 & 2192 & 1932 & 1.13 \\
BMID000000140231 & 968 & 891 & 1.09 \\
BMID000000140232 & 371 & 328 & 1.13 \\
BMID000000140233 & 3856 & 3306 & 1.17 \\
BMID000000140234 & 2527 & 2158 & 1.17 \\
BMID000000140235 & 1840 & 1589 & 1.16 \\
BMID000000140236 & 3555 & 3095 & 1.15 \\
BMID000000140237 & 1365 & 1228 & 1.11 \\
BMID000000140238 & 3960 & 3476 & 1.14 \\
BMID000000140239 & 2588 & 2197 & 1.18 \\
BMID000000140240 & 659 & 593 & 1.11 \\
BMID000000140241 & 3168 & 2763 & 1.15 \\
BMID000000140242 & 3203 & 2835 & 1.13 \\
BMID000000140243 & 3893 & 3425 & 1.14 \\
BMID000000140244 & 4325 & 3785 & 1.14 \\
BMID000000140245 & 4387 & 3858 & 1.14 \\
BMID000000140246 & 4437 & 3837 & 1.16 \\
BMID000000140247 & 4506 & 3917 & 1.15 \\
BMID000000140248 & 4156 & 3612 & 1.15 \\
BMID000000140249 & 1993 & 1797 & 1.11 \\
BMID000000140250 & 2213 & 1933 & 1.14 \\
BMID000000140251 & 3034 & 2733 & 1.11 \\
BMID000000140252 & 3374 & 2953 & 1.14 \\
BMID000000140253 & 2469 & 2069 & 1.19 \\
BMID000000140254 & 1326 & 1196 & 1.11 \\
BMID000000140255 & 3956 & 3430 & 1.15 \\
BMID000000140256 & 2665 & 2378 & 1.12 \\
BMID000000140257 & 3363 & 2823 & 1.19 \\
BMID000000140258 & 2913 & 2513 & 1.16 \\
BMID000000140259 & 2008 & 1784 & 1.13 \\
BMID000000140260 & 4608 & 3983 & 1.16 \\
BMID000000140261 & 1293 & 1187 & 1.09 \\
BMID000000140262 & 3064 & 2696 & 1.14 \\
BMID000000140263 & 3705 & 3260 & 1.14 \\
BMID000000140264 & 2435 & 2051 & 1.19 \\
BMID000000140265 & 2281 & 2066 & 1.1 \\
BMID000000140266 & 2623 & 2398 & 1.09 \\
BMID000000140267 & 1811 & 1598 & 1.13 \\
BMID000000140268 & 4321 & 3707 & 1.17 \\
BMID000000140269 & 3571 & 2983 & 1.2 \\
BMID000000140270 & 2199 & 1858 & 1.18 \\
BMID000000140271 & 3941 & 3447 & 1.14 \\
BMID000000140272 & 2354 & 2071 & 1.14 \\
BMID000000140273 & 2010 & 1789 & 1.12 \\
BMID000000140274 & 1960 & 1810 & 1.08 \\
BMID000000140275 & 3277 & 2945 & 1.11 \\
BMID000000140276 & 3333 & 2978 & 1.12 \\
BMID000000140277 & 4625 & 4030 & 1.15 \\
BMID000000140278 & 987 & 933 & 1.06 \\
BMID000000140279 & 4060 & 3641 & 1.12 \\
BMID000000140280 & 2090 & 1837 & 1.14 \\
BMID000000140281 & 2474 & 2294 & 1.08 \\
BMID000000140282 & 1667 & 1479 & 1.13 \\
BMID000000140283 & 1680 & 1504 & 1.12 \\
BMID000000140284 & 3887 & 3388 & 1.15 \\
BMID000000140285 & 3376 & 2894 & 1.17 \\
BMID000000140286 & 2752 & 2429 & 1.13 \\
BMID000000140287 & 4095 & 3580 & 1.14 \\
BMID000000140288 & 3799 & 3266 & 1.16 \\
BMID000000140289 & 4336 & 3676 & 1.18 \\
BMID000000140290 & 2041 & 1774 & 1.15 \\
BMID000000140291 & 4089 & 3578 & 1.14 \\
BMID000000140292 & 3482 & 2922 & 1.19 \\
BMID000000140293 & 3836 & 3392 & 1.13 \\
BMID000000140294 & 3880 & 3381 & 1.15 \\
BMID000000140295 & 1481 & 1339 & 1.11 \\
BMID000000140296 & 3107 & 2783 & 1.12 \\
BMID000000140297 & 3799 & 3318 & 1.14 \\
BMID000000140298 & 2358 & 2102 & 1.12 \\
BMID000000140299 & 1963 & 1700 & 1.15 \\
BMID000000140300 & 2796 & 2512 & 1.11 \\
BMID000000140301 & 1203 & 1110 & 1.08 \\
BMID000000140302 & 406 & 366 & 1.11 \\
BMID000000140303 & 3145 & 2748 & 1.14 \\
BMID000000140304 & 3740 & 3289 & 1.14 \\
BMID000000140305 & 1640 & 1502 & 1.09 \\
BMID000000140306 & 2058 & 1839 & 1.12 \\
BMID000000140307 & 2732 & 2475 & 1.1 \\
BMID000000140308 & 1648 & 1459 & 1.13 \\
BMID000000140309 & 1168 & 1082 & 1.08 \\
BMID000000140310 & 3888 & 3429 & 1.13 \\
BMID000000140311 & 1673 & 1534 & 1.09 \\
BMID000000140312 & 2826 & 2469 & 1.14 \\
BMID000000140313 & 5056 & 4428 & 1.14 \\
BMID000000140314 & 1425 & 1319 & 1.08 \\
BMID000000140315 & 1116 & 1036 & 1.08 \\
BMID000000140316 & 2138 & 1950 & 1.1 \\
BMID000000140317 & 3535 & 2972 & 1.19 \\
BMID000000140318 & 1519 & 1363 & 1.11 \\
BMID000000140319 & 2117 & 1927 & 1.1 \\
BMID000000140320 & 2531 & 2269 & 1.12 \\
BMID000000140321 & 3513 & 3071 & 1.14 \\
BMID000000140322 & 4339 & 3716 & 1.17 \\
BMID000000140323 & 597 & 541 & 1.1 \\
BMID000000140324 & 1245 & 1156 & 1.08 \\
BMID000000140325 & 2513 & 2334 & 1.08 \\
BMID000000140326 & 2607 & 2399 & 1.09 \\
BMID000000140327 & 2244 & 1930 & 1.16 \\
BMID000000140328 & 974 & 872 & 1.12 \\
BMID000000140329 & 3231 & 2880 & 1.12 \\
BMID000000140330 & 2011 & 1828 & 1.1 \\
BMID000000140331 & 1693 & 1542 & 1.1 \\
BMID000000140332 & 4269 & 3669 & 1.16 \\
BMID000000140333 & 1633 & 1509 & 1.08 \\
BMID000000140334 & 3546 & 3107 & 1.14 \\
BMID000000140335 & 1650 & 1516 & 1.09 \\
BMID000000140336 & 1928 & 1771 & 1.09 \\
BMID000000140337 & 4316 & 3703 & 1.17 \\
BMID000000140338 & 1548 & 1377 & 1.12 \\
BMID000000140339 & 1879 & 1697 & 1.11 \\
BMID000000140340 & 656 & 635 & 1.03 \\
BMID000000140341 & 2302 & 1937 & 1.19 \\
BMID000000140342 & 3103 & 2699 & 1.15 \\
BMID000000140343 & 2655 & 2402 & 1.11 \\
BMID000000140344 & 1787 & 1687 & 1.06 \\
BMID000000140345 & 3189 & 2682 & 1.19 \\
BMID000000140346 & 1921 & 1778 & 1.08 \\
BMID000000140347 & 2999 & 2675 & 1.12 \\
BMID000000140348 & 1930 & 1818 & 1.06 \\
BMID000000140349 & 2895 & 2627 & 1.1 \\
BMID000000140350 & 1799 & 1545 & 1.16 \\
BMID000000140351 & 3620 & 3170 & 1.14 \\
BMID000000140352 & 2586 & 2351 & 1.1 \\
BMID000000140353 & 2927 & 2631 & 1.11 \\
BMID000000140354 & 4744 & 4174 & 1.14 \\
BMID000000140355 & 4673 & 4017 & 1.16 \\
BMID000000140356 & 4670 & 4043 & 1.16 \\
BMID000000140357 & 1673 & 1485 & 1.13 \\
BMID000000140358 & 4382 & 3794 & 1.15 \\
BMID000000140359 & 3200 & 2852 & 1.12 \\
BMID000000140360 & 807 & 767 & 1.05 \\
BMID000000140361 & 3400 & 2970 & 1.14 \\
BMID000000140362 & 5819 & 4948 & 1.18 \\
BMID000000140363 & 1311 & 1166 & 1.12 \\
BMID000000140364 & 3185 & 2785 & 1.14 \\
BMID000000140365 & 3962 & 3454 & 1.15 \\
BMID000000140366 & 4107 & 3571 & 1.15 \\
BMID000000140367 & 3490 & 3092 & 1.13 \\
BMID000000140368 & 1738 & 1628 & 1.07 \\
BMID000000140369 & 2317 & 2004 & 1.16 \\
BMID000000140370 & 4068 & 3565 & 1.14 \\
BMID000000140371 & 4272 & 3782 & 1.13 \\
BMID000000140372 & 2109 & 1834 & 1.15 \\
BMID000000140373 & 1259 & 1137 & 1.11 \\
BMID000000140374 & 2952 & 2547 & 1.16 \\
BMID000000140375 & 944 & 892 & 1.06 \\
BMID000000140376 & 828 & 788 & 1.05 \\
BMID000000140377 & 2595 & 2370 & 1.09 \\
BMID000000140378 & 4528 & 3864 & 1.17 \\
BMID000000140379 & 4014 & 3519 & 1.14 \\
BMID000000140380 & 1609 & 1484 & 1.08 \\
BMID000000140381 & 4379 & 3779 & 1.16 \\
BMID000000140382 & 1769 & 1563 & 1.13 \\
BMID000000140383 & 2365 & 2009 & 1.18 \\
BMID000000140384 & 3926 & 3477 & 1.13 \\
BMID000000140385 & 3510 & 3099 & 1.13 \\
BMID000000140386 & 4133 & 3579 & 1.15 \\
BMID000000140387 & 3096 & 2779 & 1.11 \\
BMID000000140388 & 2010 & 1791 & 1.12 \\
BMID000000140389 & 3635 & 3187 & 1.14 \\
BMID000000140390 & 2416 & 2150 & 1.12 \\
BMID000000140391 & 2861 & 2535 & 1.13 \\
BMID000000140392 & 3013 & 2708 & 1.11 \\
BMID000000140393 & 1659 & 1463 & 1.13 \\
BMID000000140394 & 3147 & 2770 & 1.14 \\
BMID000000140395 & 3317 & 2908 & 1.14 \\
BMID000000140396 & 2958 & 2649 & 1.12 \\
BMID000000140397 & 2022 & 1792 & 1.13 \\
BMID000000140398 & 2715 & 2417 & 1.12 \\
BMID000000140399 & 2589 & 2203 & 1.18 \\
BMID000000140400 & 2765 & 2445 & 1.13 \\
BMID000000140401 & 3418 & 3017 & 1.13 \\
BMID000000140402 & 2979 & 2680 & 1.11 \\
BMID000000140403 & 3301 & 2907 & 1.14 \\
BMID000000140404 & 3586 & 3000 & 1.2 \\
BMID000000140405 & 1935 & 1820 & 1.06 \\
BMID000000140406 & 2768 & 2448 & 1.13 \\
BMID000000140407 & 2771 & 2484 & 1.12 \\
BMID000000140408 & 4011 & 3375 & 1.19 \\
BMID000000140409 & 3853 & 3397 & 1.13 \\
BMID000000140410 & 2787 & 2531 & 1.1 \\
BMID000000140411 & 3029 & 2651 & 1.14 \\
BMID000000140412 & 4639 & 3967 & 1.17 \\
BMID000000140413 & 1939 & 1668 & 1.16 \\
BMID000000140414 & 2805 & 2528 & 1.11 \\
BMID000000140415 & 1289 & 1181 & 1.09 \\
BMID000000140416 & 1608 & 1422 & 1.13 \\
BMID000000140417 & 3099 & 2768 & 1.12 \\
BMID000000140418 & 2859 & 2603 & 1.1 \\
BMID000000140419 & 2059 & 1787 & 1.15 \\
BMID000000140420 & 3833 & 3330 & 1.15 \\
BMID000000140421 & 3042 & 2756 & 1.1 \\
BMID000000140422 & 2131 & 1843 & 1.16 \\
BMID000000140423 & 4512 & 3900 & 1.16 \\
BMID000000140424 & 1711 & 1545 & 1.11 \\
BMID000000140425 & 3729 & 3235 & 1.15 \\
BMID000000140426 & 1176 & 1086 & 1.08 \\
BMID000000140427 & 2551 & 2160 & 1.18 \\
BMID000000140428 & 2253 & 1935 & 1.16 \\
BMID000000140429 & 2765 & 2491 & 1.11 \\
BMID000000140430 & 3734 & 3351 & 1.11 \\
BMID000000140431 & 1276 & 1184 & 1.08 \\
BMID000000140432 & 3914 & 3395 & 1.15 \\
BMID000000140433 & 2725 & 2362 & 1.15 \\
BMID000000140434 & 4294 & 3661 & 1.17 \\
BMID000000140435 & 4395 & 3765 & 1.17 \\
BMID000000140436 & 2958 & 2614 & 1.13 \\
BMID000000140437 & 2704 & 2474 & 1.09 \\
BMID000000140438 & 3824 & 3391 & 1.13 \\
BMID000000140439 & 2996 & 2686 & 1.12 \\
BMID000000140440 & 2371 & 2172 & 1.09 \\
BMID000000140441 & 1848 & 1743 & 1.06 \\
BMID000000140442 & 2732 & 2484 & 1.1 \\
BMID000000140443 & 2627 & 2373 & 1.11 \\
BMID000000140444 & 3027 & 2639 & 1.15 \\
BMID000000140445 & 4260 & 3706 & 1.15 \\
BMID000000140446 & 3733 & 3311 & 1.13 \\
BMID000000140447 & 4005 & 3519 & 1.14 \\
BMID000000140448 & 2114 & 1892 & 1.12 \\
BMID000000140449 & 4333 & 3708 & 1.17 \\
BMID000000140450 & 4198 & 3731 & 1.13 \\
BMID000000140451 & 3114 & 2729 & 1.14 \\
BMID000000140452 & 4337 & 3737 & 1.16 \\
BMID000000140453 & 2492 & 2196 & 1.13 \\
BMID000000140454 & 5072 & 4335 & 1.17 \\
BMID000000140455 & 4051 & 3559 & 1.14 \\
BMID000000140456 & 2778 & 2513 & 1.11 \\
BMID000000140457 & 1753 & 1521 & 1.15 \\
BMID000000140458 & 3846 & 3369 & 1.14 \\
BMID000000140459 & 2545 & 2290 & 1.11 \\
BMID000000140460 & 4547 & 4057 & 1.12 \\
BMID000000140461 & 3337 & 2961 & 1.13 \\
BMID000000140462 & 389 & 347 & 1.12 \\
BMID000000140463 & 4895 & 4216 & 1.16 \\
BMID000000140464 & 1078 & 1030 & 1.05 \\
BMID000000140465 & 3114 & 2791 & 1.12 \\
BMID000000140466 & 3546 & 2963 & 1.2 \\
BMID000000140467 & 4355 & 3745 & 1.16 \\
BMID000000140468 & 4418 & 3823 & 1.16 \\
BMID000000140469 & 3563 & 3189 & 1.12 \\
BMID000000140470 & 4095 & 3573 & 1.15 \\
BMID000000140471 & 3551 & 3209 & 1.11 \\
BMID000000140472 & 1743 & 1555 & 1.12 \\
BMID000000140473 & 4040 & 3446 & 1.17 \\
 \hline
\textbf{Average:} & 2879 & 2532 & 1.14 \\
\hline
\end{longtable}
\end{center}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- A video tutorial for Mimoza.}
  The video shows the main features of Mimoza, on the example of the map for the yeast consensus genome-scale metabolic network~\cite{Herrgard2008}.


\end{backmatter}
\end{document}
